build.gradle

```diff
@@ -109,12 +110,48 @@ task sourcesJar(type: Jar) {
     from sourceSets.main.allJava
 }
 
+task javadocJar(type: Jar) {
+   classifier = 'javadoc'
+   from javadoc.destinationDir
+}
+
 publishing {
     publications {
         shadow(MavenPublication) { publication ->
             project.shadow.component(publication)
             artifact sourcesJar
         }
+        sonatype(MavenPublication) { publication ->
```

Comment:
we may not need the sonatype publication defined separately at this time. Move everything inside `sonatype(MavenPublication) { publication ->`  within `shadow(MavenPublication) { publication ->`

Reply:
For now. `shadow` is not signing. `sonatype` is signing.
If I move `sonatype` into shadow, then for our Jenkins job, it will try to do signing. It will fail because key is not on our Jenkins host.

Reply:
You are right, we don't want to sign artifacts unless they are being published to sonatype. One option we can try is to add a condition to the signing step - this way define only one publication but signing only occurs if we are going to publish to sonatype. Let me know if this works:

```
signing {
    required { gradle.taskGraph.hasTask("publishShadowPublicationToSonatype-stagingRepository") }
    sign publishing.publications.shadow
}
```